envision: autopatchelf after build

Pandapip1 opened the pull request:
## Description of changes

Adds an OpenXR document to establish an expectation (and a little bit of responsibility), and updates envision to align with that new document. This is necessary because otherwise all OpenXR applications would need to run in a fhsenv with a list of commonly-used provider dependencies.

Also makes envision align with said document.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
nixos/iso-image: fix isoImage.grubTheme = null; logic

NetaliDev committed:
zammad: refactor package

NetaliDev committed:
nixos/zammad: refactor module

NetaliDev committed:
nixos/tests/zammad: refactor test

Pandapip1 committed:
envision: autopatchelf after build

Pandapip1 committed:
doc/interoperability: add OpenXR doc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4485

rhelmot committed:
freebsd: improve overridability of entire package set

- remove duplicated patches directory attribute
- make several version number checks >= or <= instead of ==
- add kernel derivation parameters to adjust build config

Pandapip1 commented:
I'm aware `autopatchelfHook` exists, but it doesn't actually expose the `autopatchelf` binary.

Artturin commented:
> I'm aware `autopatchelfHook` exists, but it doesn't actually expose the `autopatchelf` binary.

Maybe expose it in the `passthru` of `autoPatchelfHook` if its maintainers are ok with that.

The old `autopatchelf` needs to be clearly labelled as `old-autopatchelf` to make sure no one uses it.

Pandapip1 commented:
> The old autopatchelf needs to be clearly labelled as old-autopatchelf to make sure no one uses it. (preferably it's not added to nixpkgs)

Out of curiosity, why shouldn't it be used?

Artturin commented:
> > The old autopatchelf needs to be clearly labelled as old-autopatchelf to make sure no one uses it. (preferably it's not added to nixpkgs)
> 
> Out of curiosity, why shouldn't it be used?

It's likely lacking all the fixes the in-tree auto-patchelf.sh had before being rewritten in python and the fixes the python version has

Scrumplex commented:
> Maybe expose it in the `passthru` of `autoPatchelfHook` if its maintainers are ok with that.

I have proposed #340162 for this ^^

ElvishJerricco commented:
@Pandapip1

> I'm aware autopatchelfHook exists, but it doesn't actually expose the autopatchelf binary.

Because it doesn't use it. That binary is obsolete and what we use now is in-tree.

Pandapip1 commented:
Yeah, @Scrumplex already told me that. I'm reviewing their PR to expose it.

r3n3gad3p3arl committed:
openutau: bump dotnet version 7 -> 8

Pandapip1 commented:
Tracking @Scrumplex's exposing of auto-patchelf: https://nixpk.gs/pr-tracker.html?pr=340162

dotlambda committed:
python312Packages.ocrmypdf: 16.5.0 -> 16.6.2

Diff: https://github.com/ocrmypdf/OCRmyPDF/compare/v16.5.0...v16.6.2

Changelog: https://github.com/ocrmypdf/OCRmyPDF/blob/v16.6.2/docs/release_notes.rst

Pandapip1 committed:
auto-patchelf: Fix missing defaults and add --no-add-existing option

Pandapip1 commented:
Bump. Would like to get this merged in time for 24.11

pwaller committed:
llvmPackages.compiler_rt: Fix compiler_rt version tests for git

Without these fixes, the git build currently fails because the tests are
out of date.

One is updated since it was fixed upstream, the other is still needed
for the git build, but testing `version` resulted in a test against the
`unstable-*` string and not the one with the version number in it.

Signed-off-by: Peter Waller <p@pwaller.net>

jopejoe1 committed:
vscode-extensions: set pname

pwaller committed:
llvmPackages: Make targetLlvmLibraries overridable

Pull #320261 introduced the possibility to consistently override
dependencies within an llvm package set. This is useful for development
and testing exotic configurations.

Go one step further and enable overriding targetLlvmLibraries.

This makes it possible to write an overlay such as:

```nix
overlays = [
  (self: super: {
    llvmPackages = super.llvmPackages.override (prev: {
      targetLlvmLibraries = super.targetPackages.llvmPackages.libraries // {
        compiler-rt = super.targetPackages.llvmPackages.libraries.compiler-rt.override {
          ...
        }
      };
    });
  })
];
```

... where the overridden compiler-rt will be used in a pkgsLLVM build.

As a straw man, I've done the minimally invasive thing to the code
structure: `targetLlvmLibraries` is not an explicitly named parameter
for llvmPackages; but it is available in `packageSetArgs` if passed.
This makes it slightly less discoverable, but this seems like a
reasonable tradeoff considered that modifying this would be a fairly
advanced/esoteric thing to need to do.

In some ways it would be better to have as an explicit parameter with a
default, but the obvious thing won't work because the default needs to
be a non-trivial expression. Potentially we could instead have it as a
defaulted parameter with the value of 'null', and if it's null, then
compute the current thing.

Signed-off-by: Peter Waller <p@pwaller.net>

CHN-beta committed:
stdenv: add Silvermont support, remove incorrect AES support

OPNA2608 committed:
chiptrack: 0.3.1 -> 0.5

layus commented on pkgs/by-name/au/auto-patchelf/source/auto-patchelf.py:
```diff
@@ -394,14 +413,21 @@ def main() -> None:
         help="Attempt to search for and relink libc dependencies.",
     )
+    parser.add_argument(
+        "--no-add-existing",
```
I think the name does not convey the right meaning. I would prefer something like `--no-default-paths` or `--no-self-references`.

The description below can be confusing to the readers too. The thing that is avoided here is adding the paths to patch to the search paths. Ir makes sense as you are supposed to prefer binaries that you are providing over system ones.

That is why I am a bit uncertain about this whole change. Until now this problem has always been solved by removing the libraries that autopatchelf should not consider, or by passing files in paths_to_patch.

The patch is well written, and there is no problem with it _per se_, but I would like to understand why and if it is really needed.

Pandapip1 replied:
> The patch is well written, and there is no problem with it *per se*, but I would like to understand why and if it is really needed.

The reason this is needed is that in an fhsenv (such as the envision build environment), auto-patchelf will find the libraries in the fhsenv due to the binaries' rpaths, and won't modify the binaries that need the rpaths to point into the nix store.

layus commented on pkgs/by-name/en/envision/package.nix:
```diff
@@ -1,45 +1,36 @@
 {
+  lib,
```
nit: this file does not pass the formatting check on CI

layus replied:
Would you mind fixing the formatting. I can also do it myself if you prefer.

layus reviewed (changes requested):
Thanks for pushing this forward, it is a very interesting idea that opens the way to more self-updating packages. Could you answer the questions in my above (below?) comment ? I promise I will be more responsive to interact and ultimately merge.

theotheroracle committed:
pmbootstrap: 2.3.1 > 3.0.0

Builditluc committed:
youki: add systemd build flag

Signed-off-by: Builditluc <git@builditluc.eu>

jopejoe1 committed:
aspellDicts: expose pname and version

Commit:
netbird-dashboard: 2.5.0 -> 2.7.0

NetaliDev committed:
prometheus-aruba-exporter: init at unstable-2023-01-18

utopiatopia committed:
cemu-ti: add aarch64-linux platform

9999years committed:
git-prole: 0.5.1 -> 0.5.3

Diff: https://github.com/9999years/git-prole/compare/refs/tags/v0.5.1...v0.5.3

Changelog: https://github.com/9999years/git-prole/releases/tag/v0.5.3

RossComputerGuy committed:
flutter: revert remove usages of aliases {build,host,target}Platform

PR #350299 broke Flutter's update script with how it handled getting the
platform. Furthermore, the problem was made worse in the last update to
Flutter which had to work around that problem. We'll just revert it for
now and do a proper fix once we can figure out how to satify setting the
different platforms without causing any more problems.

BonusPlay committed:
ghidra: add wasm extension

malob committed:
python312Packages.openai: 1.52.1 -> 1.54.5

Diff: https://github.com/openai/openai-python/compare/refs/tags/v1.52.1...v1.54.5

Changelog: https://github.com/openai/openai-python/releases/tag/v1.54.5

aaronjheng committed:
cue: format with nixfmt

aaronjheng committed:
cue: 0.10.1 -> 0.11.0

goatastronaut0212 committed:
unciv: 4.14.5-patch1 -> 4.14.9

r-ryantm committed:
python312Packages.python-socks: 2.5.2 -> 2.5.3

GaetanLepage committed:
tinymist: 0.12.0 -> 0.12.4

Diff:
https://github.com/Myriad-Dreamin/tinymist/compare/refs/tags/v0.12.0...0.12.4

Changelog:
https://github.com/Myriad-Dreamin/tinymist/blob/refs/tags/v0.12.4/CHANGELOG.md

r-ryantm committed:
firefox-beta-unwrapped: 133.0b1 -> 133.0b9

nicoonoclaste committed:
factorio: fix `updateScript` definition

The update script does not take `username` and `token` parameters anymore,
since 44754ac517ceecf2a8b95318c45e28192e813a8b (thanks Luke!)

As a side-effect, R. RyanTM should now keep `factorio` up-to-date
like the good bot it is.  <3

r-ryantm committed:
suitesparse-graphblas: 9.3.1 -> 9.4.2

gepbird committed:
lua-language-server: 3.13.0 -> 3.13.2

fgaz committed:
ladybird: 0-unstable-2024-11-06 -> 0-unstable-2024-11-21

techknowlogick committed:
vault: 1.18.1 -> 1.18.2

Diff: https://github.com/hashicorp/vault/compare/v1.18.1...v1.18.2

Changelog: https://github.com/hashicorp/vault/blob/v1.18.2/CHANGELOG.md

techknowlogick committed:
vault-bin: 1.18.1 -> 1.18.2

Commit:
telegram-bot-api: 7.11 -> 8.0

thtrf committed:
vscode-extensions.chanhx.crabviz: init at 0.4.0

dotlambda committed:
paperless-ngx: fix tests with OCRmyPDF 16.6

utopiatopia committed:
cemu-ti: mark x86_64-linux broken

emilazy committed:
sumokoin: drop

Abandoned cryptocurrency; no upstream activity since 2021.

thiagokokada committed:
libretro-core-info: move to pkgs/by-name, format with nixfmt-rfc-style

thiagokokada committed:
retroarch-assets: move to pkgs/by-name, format with nixfmt-rfc-style

thiagokokada committed:
retroarch-joypad-autoconfig: move to pkgs/by-name, format by nixfmt-rfc-style

thiagokokada committed:
kodi-retroarch-advanced-launchers: move to pkgs/by-name, format with nixfmt-rfc-style

thiagokokada committed:
retroarch: format with nixfmt-rfc-style

thiagokokada committed:
retroarch: add withCores helper

thiagokokada committed:
retroarch-free: init

r-ryantm committed:
wordpress: 6.7 -> 6.7.1

toastal committed:
ocamlPackages.eio: 1.1 → 1.2

tfc committed:
python3Packages.pyside2-tools: fix

emilazy committed:
dotnet-{sdk,runtime,aspnetcore}_{6,7}: mark as EOL

See <https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core>.

keenanweaver committed:
woof-doom: 14.5.0 -> 15.0.0

thiagokokada committed:
retroarch-{bare,full}: renamed from retroarch{Bare,Full}

thiagokokada committed:
retroarch-bare: add passthru.wrapper, refactor derivation to use it

thiagokokada committed:
retroarch-bare: move to pkgs/by-name

thiagokokada committed:
retroarch-bare: remove top-level `with lib` in meta

thiagokokada committed:
libretro: renamed from retroarch

thiagokokada committed:
libretro: update README.md

emilazy committed:
libbitcoin{,-client,-explorer,-network,-protocol}: drop

These are the last packages in the tree that depend on Boost
1.75. There hasn’t been a release since 2023; upstream have stated
in <https://github.com/libbitcoin/libbitcoin-system/issues/1234> that
they do not plan to maintain the current version, and are currently
working on a backwards‐incompatible major update.

Nothing in the tree uses these libraries. The only other software
distributions on Repology packaging them are YACP and the AUR, neither
of which has had an update since 2019. They were removed from Ubuntu
after 16.04, and have more recently been removed from [SlackBuilds.org]
and [Homebrew] with the same reasoning that they’re keeping Boost
1.75 around unnecessarily and no packaged software wants them.

[SlackBuilds.org]: https://github.com/SlackBuildsOrg/slackbuilds/commit/17c7d0e430eadaab2ea02d435b0f48977961eba9
[Homebrew]: https://github.com/Homebrew/homebrew-core/pull/157427

Since these packages are also unmaintained in Nixpkgs, let’s just
drop them and if anyone wants to package the 4.x version when it’s
out they can re‐add it.

emilazy committed:
boost175: drop

Obsolete version with no remaining users in the tree. Already removed
by other distributions.

emilazy committed:
rippled{,-validator-keys-tool}: drop

This package was broken from May until November’s ZHF due to a Git
security change. In itself, that’s a one‐line fix, but a greater
concern is that this package hasn’t been updated since 2022; there
have been many releases since, including backward‐incompatible
protocol changes:

> If you operate an XRP Ledger server, upgrade to version 1.10.0 by
> March 21 to ensure service continuity.

Since it is after March 2023 and nobody has reported this package
broken, I conclude that nobody uses it, and we should remove it until
an interested maintainer comes along.

ShamrockLee committed:
singularity-tools: enable __structuredAttrs and pass contents directly

john-rodewald committed:
python312Packages.pinocchio: Disable test that fails on darwin

Stunkymonkey committed:
nixos/manticore: fix mkKeyValueDefault

wegank committed:
nvc: move to pkgs/by-name

wegank committed:
nvc: reformat

wegank committed:
nvc: 1.14.1 -> 1.14.2

thiagokokada committed:
retroarch: define it using wrapRetroArch instead retroarch-bare.wrapper

thiagokokada committed:
retroarch-{bare,free}: use lib instead of builtins

thiagokokada committed:
retroarch-bare: add missing . in longDescription

mjoerg committed:
radicle-{explorer,httpd}: 0.17.0 -> 0.17.1

the plausible.io tracker is removed from upstream radicle-explorer
https://app.radicle.xyz/nodes/seed.radicle.xyz/rad:z4V1sjrXqjvFdnCUbxPFqd5p4DtH5/patches/4e78b23dfbb52492076a1bd7d316ad0a018e292e

theobori committed:
taterclient-ddnet: 8.6.1 -> 9.0.0

mjoerg committed:
skypeforlinux: 8.132.0.201 -> 8.133.0.202

frectonz committed:
sql-studio: 0.1.27 -> 0.1.32

r-ryantm committed:
qlog: 0.39.0 -> 0.40.0

mjoerg committed:
edk2: 202408.01 -> 202411

https://github.com/tianocore/edk2/releases/tag/edk2-stable202411
https://github.com/tianocore/edk2/compare/refs/tags/edk2-stable202408.01...refs/tags/edk2-stable202411

chewblacka committed:
decent-sampler: 1.12.1 -> 1.12.5

LeSuisse committed:
zizmor: 0.2.1 -> 0.5.0

Changes:
https://github.com/woodruffw/zizmor/releases/tag/v0.5.0
https://github.com/woodruffw/zizmor/releases/tag/v0.4.0
https://github.com/woodruffw/zizmor/releases/tag/v0.3.0

TudbuT committed:
spl: 0.4.0 -> 0.4.1

dotlambda committed:
radicale: 3.3.0 -> 3.3.1

Diff: https://github.com/Kozea/Radicale/compare/refs/tags/v3.3.0...v3.3.1

Changelog: https://github.com/Kozea/Radicale/blob/refs/tags/v3.3.1/CHANGELOG.md

dotlambda committed:
radicale: format with nixfmt-rfc-style

dotlambda committed:
python312Packages.radicale-infcloud: modernize

thiagokokada committed:
retroarch: use makeBinaryWrapper

r-ryantm committed:
markdown-oxide: 0.24.2 -> 0.24.3

Commit:
convbin: 3.7 -> 5.1

dotlambda committed:
python312Packages.cachecontrol: 0.14.0 -> 0.14.1

Diff: https://github.com/ionrock/cachecontrol/compare/refs/tags/v0.14.0...v0.14.1

Changelog: https://github.com/psf/cachecontrol/releases/tag/v0.14.1

Bot-wxt1221 committed:
python3Packages.globus-sdk: 3.45.0 -> 3.48.0

r-ryantm committed:
gleam: 1.6.1 -> 1.6.2

me-and committed:
cups-kyocera-3500-4500: reorder source URLs

List the URL that is least likely to silently change its target first,
to avoid future builds breaking because the vendor released a new
package using the same URL.

Luflosi committed:
home-assistant-custom-components.solax_modbus: 2024.11.1 -> 2024.11.2

https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.11.2

emilylange committed:
chromium: resolve ref to rev for blink version string

This allows us to match the version the binaries use more closely.

For example, chromedriver darwin (binary) reports the following:

~~~bash
chromedriver --version
ChromeDriver 131.0.6778.85 (3d81e41b6f3ac8bcae63b32e8145c9eb0cd60a2d-refs/branch-heads/6778@{#2285})
~~~

while on Linux, where we build chromedriver based on the chromium
derivation from source and control the string ourselves:

~~~bash
chromedriver --version
ChromeDriver 131.0.6778.85 (131.0.6778.85-refs/heads/master@{#0})
~~~

With this commit, the version string now reports:

~~~bash
chromedriver --version
ChromeDriver 131.0.6778.85 (3d81e41b6f3ac8bcae63b32e8145c9eb0cd60a2d-refs/tags/131.0.6778.85@{#0})
~~~

This may seem like a small and unimportant detail, but turns out an
internal function within chromedriver depends on the git hash.

See https://chromium.googlesource.com/chromium/src/+/131.0.6778.85/chrome/test/chromedriver/chrome/browser_info.cc#172

This caused the tests of one package (single-file-cli) that use
selenium with chromium and chromedriver to fail in 24.05.

Only in 24.05, because 24.11 and unstable removed their test dependency
on chromedriver because it wasn't available for aarch64-linux at that
time.

~~~
Running phase: checkPhase
Serving HTTP on 127.0.0.1 port 8000 (http://127.0.0.1:8000/) ...
session not created
from unknown error: unrecognized Blink revision: 131.0.6778.85 URL: http://127.0.0.1:8000
Stack: SessionNotCreatedError: session not created
from unknown error: unrecognized Blink revision: 131.0.6778.85
    at Object.throwDecodedError (/build/source/node_modules/selenium-webdriver/lib/error.js:524:15)
    at parseHttpResponse (/build/source/node_modules/selenium-webdriver/lib/http.js:601:13)
    at Executor.execute (/build/source/node_modules/selenium-webdriver/lib/http.js:529:28)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
~~~

erictapen committed:
radicale: 3.3.0 -> 3.3.1 (#358813)

lukegb committed:
factorio: fix `updateScript` definition (#357689)

mkg20001 committed:
qlog: 0.39.0 -> 0.40.0 (#358767)

emilazy committed:
dotnet-{sdk,runtime,aspnetcore}_{6,7}: mark as EOL (#358533)

Mic92 committed:
nixVersions.{nix_2_20,nix_2_21,nix_2_22,nix_2_23}: drop

RossComputerGuy committed:
llvmPackages.compiler_rt: Fix  version tests for git (#354471)

RossComputerGuy committed:
llvmPackages: Make targetLlvmLibraries overridable (#355001)

philiptaron committed:
ghidra: add wasm extension (#349645)

mweinelt committed:
home-assistant-custom-components.solax_modbus: 2024.11.1 -> 2024.11.2 (#358971)

emilazy committed:
openutau: bump dotnet version 7 -> 8 (#340611)

gador committed:
radicle-{explorer,httpd}: 0.17.0 -> 0.17.1 (#358720)

Mic92 committed:
nixVersions.{nix_2_20,nix_2_21,nix_2_22,nix_2_23}: drop (#359024)

thiagokokada committed:
retroarch: refactor (#358405)

drupol committed:
tinymist: 0.12.0 -> 0.12.4 (#357498)

GaetanLepage committed:
typstyle: 0.12.3 -> 0.12.4

Diff: https://github.com/Enter-tainer/typstyle/compare/refs/tags/v0.12.3...v0.12.4

Changelog: https://github.com/Enter-tainer/typstyle/blob/refs/tags/v0.12.4/CHANGELOG.md

philiptaron committed:
nixos/manticore: fix mkKeyValueDefault (#358673)

philiptaron committed:
pmbootstrap: 2.3.1 > 3.0.0 (#355579)

adamcstephens committed:
edk2: 202408.01 -> 202411 (#358771)

r-ryantm committed:
python312Packages.facedancer: 3.0.4 -> 3.0.5

donovanglover committed:
convbin: 3.7 -> 5.1 (#356192)

drupol committed:
vscode-extensions.chanhx.crabviz: init at 0.4.0 (#358114)

donovanglover committed:
spl: 0.4.0 -> 0.4.1 (#358826)

philiptaron committed:
singularity-tools: enable __structuredAttrs and pass contents directly (#358723)

OPNA2608 committed:
chiptrack: 0.3.1 -> 0.5 (#355208)

r-ryantm committed:
python312Packages.pyschlage: 2024.8.0 -> 2024.11.0

donovanglover committed:
sql-studio: 0.1.27 -> 0.1.32 (#358764)

donovanglover committed:
taterclient-ddnet: 8.6.1 -> 9.0.0 (#358727)

donovanglover committed:
zizmor: 0.2.1 -> 0.5.0 (#358793)

donovanglover committed:
woof-doom: 14.5.0 -> 15.0.0 (#358579)

NickCao committed:
vault: 1.18.1 -> 1.18.2 (#358001)

NetaliDev committed:
nixos/zammad: refactor package, module and nixos-test (#277456)

jopejoe1 committed:
vscode-extensions: set pname (#354740)

jopejoe1 committed:
aspellDicts: expose pname and version (#356318)

NetaliDev committed:
prometheus-aruba-exporter: init at unstable-2023-01-18 (#356827)

philiptaron committed:
stdenv: add Silvermont support, remove incorrect AES support (#355127)

donovanglover committed:
netbird-dashboard: 2.5.0 -> 2.7.0 (#356738)

donovanglover committed:
lua-language-server: 3.13.0 -> 3.13.2 (#357849)

lf- committed:
git-prole: 0.5.1 -> 0.5.3 (#357066)

donovanglover committed:
markdown-oxide: 0.24.2 -> 0.24.3 (#358830)

nixpkgs-merge-bot[bot] committed:
suitesparse-graphblas: 9.3.1 -> 9.4.2 (#357761)

wegank committed:
python312Packages.pinocchio: Disable test that fails on darwin (#358668)

wegank committed:
sumokoin: drop (#358866)

wegank committed:
rippled: drop (#358864)

wegank committed:
wordpress: 6.7 -> 6.7.1 (#358426)

wegank committed:
nvc: 1.14.1 -> 1.14.2 (#358689)

wegank committed:
libbitcoin{,-client,-explorer,-network,-protocol}: drop; boost175: drop (#358867)

wegank committed:
youki: add systemd build flag (#355607)

kirillrdy committed:
cemu-ti: platform changes (#356874)

drupol committed:
typstyle: 0.12.3 -> 0.12.4 (#359056)

NickCao committed:
python312Packages.facedancer: 3.0.4 -> 3.0.5 (#359067)

NickCao committed:
python312Packages.pyschlage: 2024.8.0 -> 2024.11.0 (#359071)

RossComputerGuy committed:
flutter: revert remove usages of aliases {build,host,target}Platform (#357173)

happysalada committed:
python312Packages.openai: 1.52.1 -> 1.54.5 (#354068)

happysalada committed:
gleam: 1.6.1 -> 1.6.2 (#358888)

dotlambda committed:
python312Packages.ocrmypdf: 16.5.0 -> 16.6.2 (#352270)

dotlambda committed:
python312Packages.cachecontrol: 0.14.0 -> 0.14.1 (#354297)

jopejoe1 committed:
python312Packages.python-socks: 2.5.2 -> 2.5.3 (#357461)

jopejoe1 committed:
unciv: 4.14.5-patch1 -> 4.14.9 (#357456)

jopejoe1 committed:
cue: 0.10.1 -> 0.11.0 (#357449)

jopejoe1 committed:
firefox-beta-unwrapped: 133.0b1 -> 133.0b9 (#357537)

emilylange committed:
chromium: resolve ref to rev for blink version string (#358997)

dotlambda committed:
flare-signal: 0.15.2 -> 0.15.6 (#359047)

Diff: https://gitlab.com/schmiddi-on-mobile/flare/-/compare/0.15.2...0.15.6

Changelog: https://gitlab.com/schmiddi-on-mobile/flare/-/blob/0.15.6/CHANGELOG.md

jopejoe1 committed:
python3Packages.globus-sdk: 3.45.0 -> 3.48.0 (#358880)

jopejoe1 committed:
decent-sampler: 1.12.1 -> 1.12.5 (#358774)

jopejoe1 committed:
skypeforlinux: 8.132.0.201 -> 8.133.0.202 (#358741)

Artturin committed:
nixos/iso-image: fix isoImage.grubTheme = null; logic (#156754)

jopejoe1 committed:
telegram-bot-api: 7.11 -> 8.0 (#358021)

Ericson2314 committed:
freebsd: improve overridability of entire package set (#339912)

jopejoe1 committed:
ladybird: 0-unstable-2024-11-06 -> 0-unstable-2024-11-21 (#357995)

Pandapip1 commented:
Sorry all; realized my mistake right as I hit enter on the force push.
